src/postgres/commands/copyConnectionString.ts

```diff
@@ -17,13 +15,17 @@ export async function copyConnectionString(context: IActionContext, node: Postgr
         node = <PostgresDatabaseTreeItem>await ext.tree.showTreeItemPicker(PostgresDatabaseTreeItem.contextValue, context);
     }
 
-    const clientConfig: ClientConfig = await checkAuthentication(context, node);
+    await checkAuthentication(context, node);
 
-    const user: string = nonNullProp(clientConfig, 'user');
-    const password: string = nonNullProp(clientConfig, 'password');
-    const host: string = nonNullProp(clientConfig, 'host');
-    const port: string = String(nonNullProp(clientConfig, 'port'));
-    const connectionString = `postgres://${user}:${password}@${host}:${port}/'${node.databaseName}'`;
+    const parsedConnectionString = node.parent.connectionString;
+
+    if (!(parsedConnectionString.username && parsedConnectionString.password)) {
+        await node.parent.updateConnectionString();
+    }
+
+    const encodedDatabaseName = encodeURIComponent(node.databaseName);
+    let connectionString: string = parsedConnectionString.getEncodedConnectionString();
```

Comment:
The method is called `getEncodedConnectionString`, so it should encode the whole connection string including database name